testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -6680,3 +6688,1393 @@ fn test_flash_block_skip_tenure() {
+    );
+
+    // check sortdb - vote threshold met (>=50% of stacked stx involved in vote) - curr_exit_proposal should be set
+    let sort_db = btc_regtest_controller.sortdb_ref();
+    let stacks_tip = SortitionDB::get_canonical_stacks_chain_tip_hash(sort_db.conn()).unwrap();
+    let stacks_block_id = StacksBlockId::new(&stacks_tip.0, &stacks_tip.1);
+    let exit_rc_info = SortitionDB::get_exit_at_reward_cycle_info(sort_db.conn(), &stacks_block_id)
+        .unwrap()
+        .unwrap();
+    assert_eq!(exit_rc_info.curr_exit_proposal, Some(53));
+    assert_eq!(exit_rc_info.curr_exit_at_reward_cycle, Some(49));
+
+    // veto the proposal with the miner_account for the next reward cycle, but only 4 times
+    let mut nonce = 457;
+    for _ in 0..4 {
+        let veto_tx = make_contract_call(
+            &miner_privk,
+            nonce,
+            260,
+            &StacksAddress::from_string("ST000000000000000000002AMW42H").unwrap(),
+            "exit-at-rc",
+            "veto-exit-rc",
+            &[Value::UInt(53)],
+        );
+
+        // okay, let's push that veto transaction!
+        submit_tx(&http_origin, &veto_tx);
+        nonce += 2;
+
+        next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+        sort_height = channel.get_sortitions_processed();
+    }
+
+    // mine blocks until the start of the next reward cycle
+    while sort_height < ((44 * pox_constants.reward_cycle_length) + 1).into() {
+        next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+        sort_height = channel.get_sortitions_processed();
+        eprintln!("Sort height: {}", sort_height);
+    }
+
+    // check sortdb - veto threshold not met - curr_exit_proposal should be set
+    let sort_db = btc_regtest_controller.sortdb_ref();
+    let stacks_tip = SortitionDB::get_canonical_stacks_chain_tip_hash(sort_db.conn()).unwrap();
+    let stacks_block_id = StacksBlockId::new(&stacks_tip.0, &stacks_tip.1);
+    let exit_rc_info = SortitionDB::get_exit_at_reward_cycle_info(sort_db.conn(), &stacks_block_id)
+        .unwrap()
+        .unwrap();
+    assert_eq!(exit_rc_info.curr_exit_proposal, None);
+    assert_eq!(exit_rc_info.curr_exit_at_reward_cycle, Some(53));
+
+    // mine blocks; node should exit in this loop
+    while sort_height < ((55 * pox_constants.reward_cycle_length) + 1).into() {
+        next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+        sort_height = channel.get_sortitions_processed();
+        eprintln!("Sort height: {}", sort_height);
+
+        // The run loop should terminate at RC 53
+        if !btc_regtest_controller.should_keep_running() {
+            info!("about to terminate");
+            test_observer::clear();
+            info!("stop cc");
+            return;
+        }
+    }
+
+    panic!("Should not have reached this line; node should exit once it gets to RC 53");
```

Comment:
why does this test end this way?

Reply:
Since the node exits, the test itself should exit (with exit code 0) at reward cycle 53. Thus, this panic should never be reached.